### ISSUE TITLE
[Reader] Some analytics events are being tracked twice

### ISSUE THREAD

**Issue by RenanLukas** [wordpress-mobile/WordPress-Android#20006]
See [here](https://github.com/wordpress-mobile/WordPress-Android/pull/19960#discussion_r1462917167)

**Comment by RenanLukas** [comment]
Hey @daniloercoli  👋 

I wanted to share some findings about this ticket after investigating [the events you've  mentioned](https://github.com/wordpress-mobile/WordPress-Android/pull/19960#discussion_r1462917167).

> When I tap on the Reader Tab the first time
> I see 2 distinct reader_filter_sheet_item_selected events tracked in analytics.

I wasn't able to reproduce this one anymore. I suppose our Reader IA changes have fixed it.

> Rotate the device
Every time i rotate the device i see 2 new reader_filter_sheet_item_selected events, together with deep_link_not_default_handler event, bumped to analytics.

I also wasn't able to reproduce this one: I'm seeing the `deep_link_not_default_handler` event (even though I'm not sure what it represents) but `reader_filter_sheet_item_selected` is only being tracked once.

> Open the single post details view
The event reader_article_rendered is tracked multiple times.

At first I thought this was a bug, but after understanding what the event `reader_article_rendered` is actually tracking I realized this was done on purpose. This event is tracked when a post details screen is created (not when it's shown) and the event `reader_article_opened` is the one that is sent when the post details is actually seen by the user. `reader_article_rendered` is sent multiple times because there are scenarios in the Reader where it loads multiple post details Fragments in a ViewPager (e.g. try opening a post details from "Subscriptions" and swiping to the left). By default one page on each side of the selected post is created (default ViewPager behavior) so the `rendered` event is sent.  The `reader_article_opened` is only sent when the post details is actually shown/selected.

I'm not sure why we need the event `reader_article_rendered`.

> Going back to the reader from the Single post details view
I see the event reader_accessed is tracked again. Not sure we'd like to bump this again here.

I believe this might be a naming problem (`_accessed` instead of `_shown`) but isn't it interesting if we have an event that tells us that the Reader "main" screen was shown? I'm thinking that it might be useful to create funnels representing certain scenarios. Perhaps we're missing an event that represents the Reader tab in the bottom navigation was tapped. WDYT?

**Comment by daniloercoli** [comment]
> I believe this might be a naming problem (`_accessed` instead of `_shown`) but isn't it interesting if we have an event that tells us that the Reader "main" screen was shown? I'm thinking that it might be useful to create funnels representing certain scenarios. Perhaps we're missing an event that represents the Reader tab in the bottom navigation was tapped. WDYT?

I believe this is the same naming issue we discussed a week ago, related to the other ticket about analytics. It's possible I opened both tickets on the same day. As we concluded last week, it's best to maintain the current naming convention and logic since it applies across different screens, not just the Reader. Additionally, we often track unique users, and even if we log multiple `reader_accessed` for the same user, it's only counted once.

Regarding whether we need a new event to track when the Reader tab is tapped in the bottom toolbar, I'm uncertain. For users who frequently use the feature, the Reader tab is likely already open in their app, so we would need to address this in the logic..anyway, I'm open to discussing this further at our next team meeting.
